runtime/syntax/java.vim

```diff
@@ -341,8 +321,6 @@ if exists("java_highlight_debug")
   syn region  javaDebugStrTemplEmbExp	contained matchgroup=javaDebugStrTempl start="\\{" end="}" contains=javaComment,javaLineComment,javaDebug\%(Paren\)\@!.*
   syn region  javaDebugStrTempl		contained start=+\%(\.[[:space:]\n]*\)\@<="+ end=+"+ contains=javaDebugStrTemplEmbExp,javaDebugSpecial
   syn region  javaDebugStrTempl		contained start=+\%(\.[[:space:]\n]*\)\@<="""[ \t\x0c\r]*$+hs=e+1 end=+"""+he=s-1 contains=javaDebugStrTemplEmbExp,javaDebugSpecial,javaDebugTextBlockError
-  " The next line is commented out, it can cause a crash for a long line
```

Comment:
There was never a test suite for it so I wouldn't know.

Apparently, `javaStringError` became a burden [sometime](https://github.com/vim/vim-history/commit/c1525e06801a41cd7e1a0927b0cc0fa62ffa2682#diff-0fddd2a7f1b39c711eb9cf31da1cd6e9ca133ed96213fedf4ae46521f1796d37) around  
October 2000, and `javaDebugStringError` was [just recently](https://github.com/vim/vim/commit/100a665634a0a2edbc755daf453c4915800684dd)  
still its copy squirreled away.

Anyway, sooner or later any fledgeling Javaist would learn  
that:

> It is a compile-time error for a line terminator (§3.4) to  
> appear after the opening " and before the matching closing  
> ". 

and that no line splicing is supported for either `"a"` or  
`'a'`:

> The line continuation escape sequence can appear in a text  
> block, but cannot appear in a character literal or a string  
> literal because each disallows a LineTerminator. 

(JLS-21, [§3.10.5](https://docs.oracle.com/javase/specs/jls/se21/html/jls-3.html#jls-3.10.5), [§3.10.7](https://docs.oracle.com/javase/specs/jls/se21/html/jls-3.html#jls-3.10.7).)

Let's toss it out.